spec/controllers/blocks_controller_spec.rb

```diff
@@ -0,0 +1,198 @@
+      expect(flash[:error][:array]).to be_present
+      expect(assigns[:block].blocked_user).to be_nil
+    end
+
+    it "requires users not be the same" do
+      user = create(:user)
+      login_as(user)
+      post :create, params: { block: { blocked_user_id: user.id } }
+      expect(response).to render_template('new')
+      expect(flash[:error][:message]).to eq("User could not be blocked.")
+      expect(flash[:error][:array]).to be_present
+      expect(flash[:error][:array]).to include("User cannot block themself")
+    end
+
+    it "succeeds" do
+      login
+      post :create, params: { block: { blocked_user_id: create(:user).id } }
+      expect(response).to redirect_to(blocks_url)
+    end
+
+    it "sets expected variables" do
+      blocker = create(:user)
+      blockee = create(:user)
+      login_as(blocker)
+      post :create, params: { block: { blocked_user_id: blockee.id, block_interactions: false, hide_them: Block::POSTS, hide_me: Block::ALL } }
+      expect(response).to redirect_to(blocks_url)
+      block = assigns(:block)
+      expect(block).not_to be_nil
+      expect(block.blocking_user).to eq(blocker)
+      expect(block.blocked_user).to eq(blockee)
+      expect(block.block_interactions).to eq(false)
+      expect(block.hide_them).to eq(Block::POSTS)
+      expect(block.hide_me).to eq(Block::ALL)
+    end
+  end
+
+  describe "GET edit" do
+    it "requires permission" do
+      block = create(:block)
+      login
+      get :edit, params: { id: block.id }
+      expect(response).to redirect_to(blocks_url)
+      expect(flash[:error]).to eq("You do not have permission to edit that block.")
+    end
+
+    it "requires valid block" do
+      login
+      get :edit, params: { id: -1 }
+      expect(response).to redirect_to(blocks_url)
+      expect(flash[:error]).to eq("Block could not be found.")
+    end
+
+    it "succeeds" do
+      block = create(:block)
+      login_as(block.blocking_user)
+      get :edit, params: { id: block.id }
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe "PUT update" do
+    it "requires permission" do
+      block = create(:block)
+      login
+      put :update, params: { id: block.id }
+      expect(response).to redirect_to(blocks_url)
+      expect(flash[:error]).to eq("You do not have permission to edit that block.")
+    end
+
+    it "requires valid block" do
+      login
+      put :update, params: { id: -1 }
+      expect(response).to redirect_to(blocks_url)
+      expect(flash[:error]).to eq("Block could not be found.")
+    end
+
+    it "requires valid variables" do
+      block = create(:block)
+      login_as(block.blocking_user)
+      put :update, params: { id: block.id, block: { hide_them: -1 } }
+      expect(response).to render_template('edit')
+      expect(flash[:error][:message]).to eq("Block could not be saved.")
+      expect(flash[:error][:array]).to be_present
+    end
+
+    it "succeeds" do
```

Comment:
Is this split from the test below intentionally? I'd prefer if they could be merged, if they're testing the same thing (that successful changes do particular things), since as is, this test isn't doing much – in the real world, nobody will submit an update with no specified data.